src/eva/core/callbacks/writers/embeddings.py

```diff
@@ -122,38 +125,105 @@ def _get_embeddings(self, prediction: torch.Tensor) -> torch.Tensor:
         with torch.no_grad():
             return self._backbone(prediction)
 
-    def _construct_save_name(self, input_name, metadata, local_idx):
-        group_name = metadata[self._group_key][local_idx] if self._group_key else None
-        save_name = os.path.splitext(input_name)[0] + ".pt"
-        if group_name:
-            save_name = os.path.join(group_name, save_name)
-        return input_name, save_name
+    def _get_item_metadata(
+        self, metadata: Dict[str, Any] | None, local_idx: int
+    ) -> Dict[str, Any] | None:
+        """Returns the metadata for the item at the given local index."""
+        if not metadata:
+            if self._metadata_keys:
+                raise ValueError("Metadata keys are provided but the batch metadata is empty.")
+            else:
+                return None
+
+        item_metadata = {}
+        for key in self._metadata_keys:
+            if key not in metadata:
+                raise KeyError(f"Metadata key '{key}' not found in the batch metadata.")
+            item_metadata[key] = metadata[key][local_idx]
+
+        return item_metadata
 
 
 def _process_write_queue(
-    write_queue: multiprocessing.Queue, output_dir: str, overwrite: bool = False
+    write_queue: multiprocessing.Queue,
+    output_dir: str,
+    metadata_keys: List[str],
+    save_every_n: int,
+    overwrite: bool = False,
 ) -> None:
-    manifest_file, manifest_writer = _init_manifest(output_dir, overwrite)
+    manifest_file, manifest_writer = _init_manifest(output_dir, metadata_keys, overwrite)
+
+    save_name_to_items: Dict[str, ITEM_DICT_ENTRY] = {}
```

Comment:
not sure if `save_name_to_items` is the best name. It suggests you "save a name to a list of items.."

Can't think of a very good name, but how about `items_dict` or `items_to_save`?

Reply:
I like to use variable names in dictionaries that describe the actual mapping, but indeed the `save_` prefix makes it confusing here. What about `name_to_items`?

Reply:
thats definitely better. Do we follow the <key>_to_<value> naming pattern elsewhere in eva?

Reply:
hehe good question, at least I mostly try to use the `to` pattern :D

Reply:
done